usr/sbin/cloudguest-lic-watcher

```diff
@@ -34,29 +35,69 @@ def has_license_changed(license_type):
             return license_type != old_license
     else:
         update_license_cache(license_type)
+        maybe_drop_registration(license_type)
+        maybe_register_system(license_type)
 
 
-utils.start_logging()
-service_name = 'guestregister.service'
-current_flavor = inst_flvr_utils.check_payg_byos()[0]
-if has_license_changed(current_flavor):
-    if current_flavor == 'BYOS' and not utils.uses_rmt_as_scc_proxy():
-        logging.info('Detected flavor change to BYOS, clean up registration')
-        utils.clean_all()
-        utils.exec_subprocess(['systemctl', 'disable', service_name])
+def maybe_drop_registration(license_type):
+    """Clean up registration data if needed"""
+    if current_flavor == 'BYOS':
+        if not utils.get_current_smt():
+            # There is no target registration server, nothing to do
+            return
+        if not utils.uses_rmt_as_scc_proxy():
+            logging.info(
+                'Detected flavor change to BYOS, clean up registration')
+            utils.clean_all()
+            utils.exec_subprocess(['systemctl', 'disable', service_name])
```

Comment:
same for `service_name`, it is set in line 96 but not in the call in line 38

Reply:
Created a constant